### ISSUE TITLE
What is the difference between @junobuild/core and @junobuild/core-peer

### ISSUE THREAD

**Issue by CodingFu** [junobuild/juno-js#155]
Hello

I have a question:

what is the difference between

https://www.npmjs.com/package/@junobuild/core

and

https://www.npmjs.com/package/@junobuild/core-peer

**Comment by peterpeterparker** [comment]
Hi @CodingFu,

Feature-wise, both `@junobuild/core` and `@junobuild/core-peer` are identical. In fact, the code of the first is copied into the second at build time.

Originally and in the future, I want to support only `@junobuild/core`, which pre-packages everything needed by the client.

Unfortunately, because the Agent-js libraries require some NodeJS polyfills, I started noticing issues with certain bundlers and frameworks, for example, SvelteKit. For some reason, even though everything is pre-polyfilled in the core, the library started to face issues with unresolved NodeJS APIs such as Buffer and other components. Despite trying hard to find a solution, I ultimately figured out that there was no way to embed Agent-js for those types of projects. The only solution was to ship the library with everything `@dfinity/agent|client|etc.` set as peer dependencies. That's why the core-peer library was added to the stack.

Note that it is not directly documented which library should be used with what type of bundler and framework, but all the guides in the documentation and templates of the create CLI are using the appropriate library type.

Let me know if you have more questions on that matter.
